### PR TITLE
ENH: Support dual index demultiplexing

### PR DESCRIPTION
**Pull request by thermokarst**

### CODE REVIEW

**Comment by thermokarst** [comment]
I can't actually get the base functionality in cutadapt to work --- opened issue https://github.com/marcelm/cutadapt/issues/378 in search of support.

**Comment by thermokarst** [comment]
Good news, `cutadapt` 2.3 was released this morning, and it includes a fix for the problems mentioned above. I just tested this PR out with 2.3 and everything seems to be working as expected.

File: q2_cutadapt/_demux.py [hunk 1]
```diff
@@ -98,8 +105,18 @@ def _write_empty_fastq_to_mux_barcode_in_seq_fmt(seqs_dir_fmt):
...
 
-def _demux(seqs, barcodes, error_tolerance, mux_fmt, batch_size):
-    barcodes = barcodes.to_series()
+def _demux(seqs, forward_barcodes, reverse_barcodes, error_tolerance,
+           mux_fmt, batch_size):
+    fwd_barcode_name = forward_barcodes.name
+    barcodes = forward_barcodes.to_series().to_frame()
+    if reverse_barcodes is not None:
+        if len(barcodes) != len(reverse_barcodes.ids):
+            raise ValueError('The number of forward barcodes does not match '
```
Comment (ChrisKeefe) [inline q2_cutadapt/_demux.py]:
```suggestion
            raise ValueError('The following samples do not have both forward and reverse barcodes: ' ...
```

**Comment by thermokarst** [comment]
Thanks @ChrisKeefe  - addressed your comments in 77f0454.

**Pull request merged**